### DISCUSSION TITLE
How to do a bulk UPSERT without specifying all columns?

### DISCUSSION THREAD

**Discussion by ewan-escience** [PostgREST/postgrest#2151] [Q&A] [answered]
Sometimes I want to update one or a few of the columns for multiple rows. I want to specify which rows to update by specifying their primary column(s) in the request body. This does not seem to be possible now. 

It seems that PATCH will update all rows that match the filtering (which is all rows, since I don't use filtering)

For PUT, you can only update one row by specifying its primary keys in the query. 

And for POST, all missing columns seem to be interpreted as null. I want unspecified columns to remain as they were.

If there is already a way to do this, please let me know. Otherwise, can this be implemented?

**Comment by steve-chavez** [comment, marked as answer]
> Otherwise, can this be implemented?

Yes, this is being tracked at https://github.com/PostgREST/postgrest/issues/1959
